pkgs/shells/bash/5.nix

```diff
@@ -88,6 +88,10 @@ stdenv.mkDerivation rec {
     "bash_cv_termcap_lib=libncurses"
   ] ++ lib.optionals (stdenv.hostPlatform.libc == "musl") [
     "--disable-nls"
+  ] ++ lib.optionals stdenv.hostPlatform.isFreeBSD [
+    "bash_cv_getenv_redef=no"
+  ] ++ lib.optionals (stdenv.hostPlatform.isFreeBSD && stdenv.buildPlatform != stdenv.hostPlatform) [
+    "bash_cv_dev_fd=absent"
```

Comment:
FreeBSD doesn't have /dev/fd?

Reply:
there is probably some extra step required

https://forums.freebsd.org/threads/needless-requirement-for-running-bash.50214/

Reply:
Ah, I see, it's an optional filesystem.

Reply:
In that case I think probably we should always set this on FreeBSD, so that whether the filesystem is mounted on the build machine doesn't affect the build output?

Reply:
I think it probably affects the ability to use '<(…)` syntax.

Reply:
If that's true, then that's not suitable for nixpkgs because we use that pattern in many places 😅

Reply:
You can still use <(), it just makes a fifo instead of using /dev/fd

Reply:
Shouldn't we be setting this always on FreeBSD?  Even on a non-cross build, we wouldn't want the mountedness of this filesystem on the build machine to impurely affect how bash is built.